implementation/src/main/java/io/smallrye/mutiny/operators/AbstractUni.java

```diff
@@ -50,11 +51,12 @@
 
     @Override
     public <T2> Uni<Tuple2<T, T2>> and(Uni<T2> other) {
-        return new UniAndGroup<>(this).uni(ParameterValidation.nonNull(other, "other")).asTuple();
+        return Infrastructure.onUniCreation(
+                new UniAndGroup<>(this).uni(ParameterValidation.nonNull(other, "other")).asTuple());
     }
 
     @Override
-    public UniOr or() {
+    public UniOr<T> or() {
         return new UniOr<>(this);
```

Comment:
Did returning a new `UniOr` require `Infrastructure` usage too?

Reply:
No, because it's not an instance of Uni, it's a group.